scalalib/src/scalafmt/ScalafmtWorker.scala

```diff
@@ -21,44 +22,84 @@ private[scalafmt] class ScalafmtWorker {
 
   def reformat(input: Seq[PathRef],
                scalafmtConfig: PathRef)(implicit ctx: Ctx): Unit = {
-    val toFormat =
-      if (scalafmtConfig.sig != configSig) input
-      else
-        input.filterNot(ref => reformatted.get(ref.path).contains(ref.sig))
-
-    if (toFormat.nonEmpty) {
-      ctx.log.info(s"Formatting ${toFormat.size} Scala sources")
-      reformatAction(toFormat.map(_.path),
-                     scalafmtConfig.path)
-      reformatted ++= toFormat.map { ref =>
-        val updRef = PathRef(ref.path)
-        updRef.path -> updRef.sig
-      }
-      configSig = scalafmtConfig.sig
+    reformatAction(input, scalafmtConfig, dryRun = false)
+  }
+
+  def checkFormat(input: Seq[PathRef],
+                  scalafmtConfig: PathRef)(implicit ctx: Ctx): Result[Unit] = {
+
+    val misformatted = reformatAction(input, scalafmtConfig, dryRun = true)
+    if (misformatted.isEmpty) {
+      Result.Success(())
     } else {
-      ctx.log.info(s"Everything is formatted already")
+      val out = ctx.log.outputStream
+      for (u <- misformatted) {
+        out.println(s"${u.path}")
```

Comment:
Printing unformatted files directly to stdout plays nicely with redirection. For instance, if I want to save the list of all unformatted files to another file yet still see error messages, I could call mill in the following way:

`mill project.checkFormat > unformatted.list`